src/runtime/pickle_runner.ts

```diff
@@ -395,9 +427,69 @@ export default class PickleRunner {
         },
       })
     }
-    return await this.invokeStep(
-      testStep.pickleStep,
-      testStep.stepDefinitions[0]
+    let stepResult
+    let afterStepHooksResult
+    const beforeStepHooksResult = await this.runStepHooks(
+      this.getBeforeStepHookDefinitions()
     )
+
+    if (beforeStepHooksResult.status !== Status.FAILED) {
+      stepResult = await this.invokeStep(
+        testStep.pickleStep,
+        testStep.stepDefinitions[0]
+      )
+      afterStepHooksResult = await this.runStepHooks(
+        this.getAfterStepHookDefinitions()
+      )
+    }
+    let cumulatedStepResult = beforeStepHooksResult
+
+    if (stepResult !== undefined) {
+      cumulatedStepResult = stepResult
+      if (beforeStepHooksResult?.duration !== null) {
+        cumulatedStepResult.duration = addDurations(
+          cumulatedStepResult.duration,
+          beforeStepHooksResult.duration
+        )
+      }
+      if (afterStepHooksResult?.duration !== null) {
+        cumulatedStepResult.duration = addDurations(
+          cumulatedStepResult.duration,
+          afterStepHooksResult.duration
+        )
+      }
+    }
+    return cumulatedStepResult
```

Comment:
I think this could be restructed as

```js
const beforeStepHooksResult = /***/
let cumulatedStepResult = beforeStepHooksResult
if (beforeStepHooksResult.status !== Status.FAILED) {
  // run step
  // update cumulatedStepResult
  // run after step hooks
  // update cumulatedStepResult
}
return cumulatedStepResult;
```
I think that tightens up where variables are defined and used vs being spread across multiple if statements

Reply:
Restructured the function per suggestion.